src/assets/scripts/client/trafficGenerator/buildPreSpawnAircraft.js

```diff
@@ -1,83 +1,271 @@
+ */
+export function _calculateAltitudeAtOffset(altitudeOffsets, offsetDistance) {
+    const indexOfDistance = 0;
+    const indexOfAltitude = 1;
+    const indexOfNextAltitudeRestriction = _findIndex(altitudeOffsets, (altitudeOffset) => {
+        return altitudeOffset[indexOfDistance] >= offsetDistance;
+    });
+    let indexOfPreviousAltitudeRestriction = indexOfNextAltitudeRestriction - 1;
+
+    if (indexOfPreviousAltitudeRestriction < -1) {
+        indexOfPreviousAltitudeRestriction = altitudeOffsets.length - 1;
+    }
+
+    if (indexOfNextAltitudeRestriction < 0) { // no restrictions ahead
+        if (indexOfPreviousAltitudeRestriction < 0) { // no restrictions ahead or behind
+            return undefined;
+        }
+
+        return altitudeOffsets[indexOfPreviousAltitudeRestriction][indexOfAltitude];
+    }
+
+    if (indexOfPreviousAltitudeRestriction < 0) { // have restrictions ahead, but none behind
+        return _floor(altitudeOffsets[indexOfNextAltitudeRestriction][indexOfAltitude], -3);
+    }
+
+    const previousAltitudeRestriction = altitudeOffsets[indexOfPreviousAltitudeRestriction];
+    const nextAltitudeRestriction = altitudeOffsets[indexOfNextAltitudeRestriction];
+    const distanceBetweenRestrictions = nextAltitudeRestriction[indexOfDistance] - previousAltitudeRestriction[indexOfDistance];
+    const altitudeBetweenRestrictions = nextAltitudeRestriction[indexOfAltitude] - previousAltitudeRestriction[indexOfAltitude];
+    const distanceFromPreviousRestrictionToOffsetDistance = offsetDistance - previousAltitudeRestriction[indexOfDistance];
+    const progressBetweenRestrictions = distanceFromPreviousRestrictionToOffsetDistance / distanceBetweenRestrictions;
+    const altitudeChangeFromPreviousRestriction = altitudeBetweenRestrictions * progressBetweenRestrictions;
+
+    return _floor(previousAltitudeRestriction[indexOfAltitude] + altitudeChangeFromPreviousRestriction, -3);
+}
+
+/**
+ * Calculate the ideal spawn altitude at the given distance along the route from the spawn point
+ *
+ * @function _calculateIdealSpawnAltitudeAtOffset
+ * @param altitudeOffsets {array<array<number>>} information about location of altitude restrictions
+ * @param offsetDistance {number} distance along route at which we want to know the ideal spawn altitude
+ * @param spawnSpeed {number} airspeed at which to spawn the aircraft
+ * @param spawnAltitude {number|array<number>} altitude specified in the spawn pattern json
+ * @param totalDistance {number} distance along route from spawn point to airspace boundary
+ * @param airspaceCeiling {number} altitude of the top of our airspace
+ * @return {number} ideal spawn altitude, in feet (rounded up to nearest thousand)
+ */
+/* eslint-disable-next-line max-len */
+export function _calculateIdealSpawnAltitudeAtOffset(altitudeOffsets, offsetDistance, spawnSpeed, spawnAltitude, totalDistance, airspaceCeiling) {
```

Comment:
please remove the `eslint-disable` comment and reformat to:
```suggestion
export function _calculateIdealSpawnAltitudeAtOffset(
    altitudeOffsets, 
    offsetDistance,
    spawnSpeed, 
    spawnAltitude, 
    totalDistance,
    airspaceCeiling
) {
```

Reply:
done